NotificationHubs/src/com/windowsazure/messaging/BaseInstallation.java

```diff
@@ -0,0 +1,310 @@
+package com.windowsazure.messaging;
+
+import com.google.gson.GsonBuilder;
+
+import java.text.SimpleDateFormat;
+import java.util.*;
+
+public abstract class BaseInstallation {
+    private String installationId;
+    private String userId;
+    private NotificationPlatform platform;
```

Comment:
In our C# code bases, I was planning on just having this be an abstract getter, skipping the property altogether. Alternatively - we could make it a required parameter in the constructors here.

Reply:
This is needed for serialization/deserialization purposes as Java GSON uses private fields for the serialization instead of properties such as C#.

Reply:
Ugh, fair enough. Let's get rid of the `BaseInstallation` constructors that don't take a `NotificationPlatform` then. If an Installation can get instantiated without that info, dispatch will be a huge pain in the neck.

Reply:
That's fair, we can do that.